x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -42,45 +49,80 @@ export const useUnifiedSearch = () => {
     to: toTS,
   };
 
+  const queryToString = (query?: Query) =>
+    typeof query?.query !== 'string' ? JSON.stringify(query?.query) : query?.query;
+
+  const handleSelectedTimeRangeChange = useCallback(
+    (selectedTime: { startTime: string; endTime: string; isInvalid: boolean }) => {
+      if (selectedTime.isInvalid) {
+        return;
+      }
+      setSelectedTimeRange(selectedTime);
+    },
+    [setSelectedTimeRange]
+  );
+
   const submitFilterChange = useCallback(
     (query?: Query, dateRange?: TimeRange, filters?: Filter[]) => {
       if (filters) {
-        filterManager.setFilters(filters);
+        applyFilters(filters);
+        filterManager.setFilters([...filterManager.getFilters(), ...filters]);
+      }
+
+      if (dateRange) {
+        handleSelectedTimeRangeChange({
+          startTime: dateRange.from,
+          endTime: dateRange.to,
+          isInvalid: false,
+        });
       }
 
       setTime({
```

Comment:
should this be inside this `if` https://github.com/elastic/kibana/pull/144181/files#diff-64f73b3a64f04f71f0f61a4ab9f2152f2181e4cfc547902f895b808d645a31e5R72?

Reply:
Thanks, done ✅